corehq/apps/translations/app_translations.py

```diff
@@ -783,14 +796,34 @@ def _looks_like_markdown(str):
     def get_markdown_node(text_node_):
         return text_node_.find("./{f}value[@form='markdown']")
 
+    def get_all_value_nodes(text_node_):
```

Comment:
What about just looping through `text_node_.findall("./{f}value")`, dropping `form="default"` if you find it, and yielding the "value" nodes. Then `get_value_node()` can stick them in a list and raise XFormException if `len(node_list) > 1`?

Reply:
+1

Reply:
@kaapstorm what do you mean " dropping form="default"" ?
Just to highlight the logic again that if there is form="default" node, it needs to be sanitised.
I like how you phrased it and i see it as

in `get_all_value_nodes` (should rename this to `get_all_default_value_nodes` or `get_all_text_value_nodes`).
find all value nodes with form="default" or blank 

then in `get_value_node`
1. if more than 1 raise exception
2. sanitise if form="default"
3. return node

Reply:
Yes, although it might be simpler to sanitise inside the loop. Something like:
```python
    def get_default_value_nodes(text_node_):
        for value_node in text_node_.findall("./{f}value"):
            if 'form' not in value_node.attrib:
                yield value_node
            elif value_node.get('form') == 'default':
                value_node.attrib.pop('form')  # "default" is an invalid value
                yield value_node

    def get_value_node(text_node_):
        default_value_nodes = list(get_default_value_nodes(text_node_))
        if len(default_value_nodes) > 1:
            raise XFormException(_(
                'Text node "{}" has conflicting translations'
            ).format(text_node_.get('id')))
        if len(default_value_nodes) == 1:
            return default_value_nodes[0]
        return WrappedNode(None)
```

Reply:
so good, thanks. Will update.